framework/wazuh/core/configuration.py

```diff
@@ -798,62 +795,129 @@ def get_active_configuration(agent_id, component, configuration):
         If there's no such file or directory in agent node, or the socket cannot send the request.
     WazuhError(1116)
         If the reply from the node contains an error.
+
+    Returns
+    -------
+    dict
+        The active configuration the agent is currently using.
     """
+    sockets_json_protocol = {'remote'}
+    component_socket_mapping = {'agent': 'agent', 'agentless': 'agentless', 'analysis': 'analysis', 'auth': 'auth',
+                                'com': 'com', 'csyslog': 'csyslog', 'integrator': 'integrator',
+                                'logcollector': 'logcollector', 'mail': 'mail', 'monitor': 'monitor',
+                                'request': 'remote', 'syscheck': 'syscheck', 'wmodules': 'wmodules'}
+
     if not component or not configuration:
         raise WazuhError(1307)
 
-    components = {"agent", "agentless", "analysis", "auth", "com", "csyslog", "integrator", "logcollector", "mail",
-                  "monitor", "request", "syscheck", "wmodules"}
-
-    # checks if the component is correct
+    # Check if the component is correct
+    components = component_socket_mapping.keys()
     if component not in components:
         raise WazuhError(1101, f'Valid components: {", ".join(components)}')
 
-    sockets_path = os_path.join(common.WAZUH_PATH, "queue", "sockets")
+    def get_active_configuration_000():
```

Comment:
Is the `_000` necessary? Look for name alternatives.

Reply:
Changed to `get_active_configuration_manager`